homeassistant/components/sensor/yr.py

```diff
@@ -0,0 +1,251 @@
+                self._info = "Forecast between " + time_data[k]['@from'] \
+                    + " and " + time_data[k]['@to'] + ". "
+
+                temp_data = time_data[k]['location']
+                if self.type in temp_data and now < valid_to:
+                    if self.type == 'precipitation' and valid_from < now:
+                        self._state = temp_data[self.type]['@value']
+                        return
+                    elif self.type == 'symbol' and valid_from < now:
+                        self._state = temp_data[self.type]['@number']
+                        return
+                    elif self.type == 'temperature':
+                        self._state = temp_data[self.type]['@value']
+                        return
+                    elif self.type == 'windSpeed':
+                        self._state = temp_data[self.type]['@mps']
+                        return
+                    elif self.type == 'pressure':
+                        self._state = temp_data[self.type]['@value']
+                        return
+                    elif self.type == 'windDirection':
+                        self._state = float(temp_data[self.type]['@deg'])
+                        return
+                    elif self.type == 'humidity':
+                        self._state = temp_data[self.type]['@value']
+                        return
+                    elif self.type == 'fog':
+                        self._state = temp_data[self.type]['@percent']
+                        return
+                    elif self.type == 'cloudiness':
+                        self._state = temp_data[self.type]['@percent']
+                        return
+                    elif self.type == 'lowClouds':
+                        self._state = temp_data[self.type]['@percent']
+                        return
+                    elif self.type == 'mediumClouds':
+                        self._state = temp_data[self.type]['@percent']
+                        return
+                    elif self.type == 'highClouds':
+                        self._state = temp_data[self.type]['@percent']
+                        return
+                    elif self.type == 'dewpointTemperature':
+                        self._state = temp_data[self.type]['@value']
+                        return
+
+
+# pylint: disable=too-few-public-methods
+class YrData(object):
+    """ Gets the latest data and updates the states. """
+
+    def __init__(self, coordinates):
+        self._url = 'http://api.yr.no/weatherapi/locationforecast/1.9/?' \
+            'lat={lat};lon={lon};msl={msl}'.format(**coordinates)
+
+        self._nextrun = datetime.datetime.fromtimestamp(0)
+        self.update()
+
+    def update(self):
+        """ Gets the latest data from yr.no """
+        now = datetime.datetime.now()
+        # check if new will be available
+        if now > self._nextrun:
+            try:
+                response = urllib.request.urlopen(self._url)
```

Comment:
Sorry, only noticed this now. Could you use [`requests`](http://docs.python-requests.org/en/latest/) instead of `urllib` ?